shell/core/plugin-routes.ts

```diff
@@ -107,87 +50,86 @@ export class PluginRoutes {
         productName = productName || plugin.name;
 
         // Replace the path - removing :product and using the actual product name instead - this avoids route collisions
-        r.route.path = `/${ productName }${ r.route.path.substr(9) }`;
-        r.route.meta = r.route.meta || {};
-
-        r.route.meta.product = r.route.meta.product || productName;
-      }
-
-      // See if the route exists
-      let existing: any;
-
-      if (r.parent) {
-        const pExisting = allRoutes.findIndex((route: any) => route.name === r.parent) as any;
-        const path = `${ pExisting.path }${ r.route.path }`;
-
-        // TODO: Validate
-        existing = allRoutes.findIndex((route: any) => route.path === path);
-      } else {
-        // no parent route
-        existing = allRoutes.findIndex((route: any) => route.name === r.route.name);
-      }
-
-      if (existing >= 0) {
-        const existingRoute = allRoutes[existing];
-
-        // Store the route so we can restore it on uninstall
-        if (plugin && existingRoute?.name) {
-          if (!this.replacedRoutes[existingRoute.name]) {
-            this.replacedRoutes[existingRoute.name] = [];
-          }
-
-          this.replacedRoutes[existingRoute.name].unshift({
-            plugin: plugin.id,
-            route:  existingRoute
-          });
-        }
+        r.path = `/${ productName }${ r.path.substr(9) }`;
+        r.meta = r.meta || {};
 
-        allRoutes.splice(existing, 1);
+        r.meta.product = r.meta.product || productName;
       }
     });
 
-    this.updateMatcher(routes, allRoutes);
+    this.updateMatcher(newRouteInfos, allRoutes);
   }
 
   private updateMatcher(newRoutes: RouteInfo[], allRoutes: RouteConfig[]) {
     // Note - Always use a new router and replace the existing router's matching
-    // Using the existing router and adding routes to it will force nuxt middleware (specifically authenticated on default layout) to
+    // Using the existing router and adding routes to it will force nuxt middleware to
     // execute many times (nuxt middleware boils down to router.beforeEach). This issue was seen refreshing in a harvester cluster with a
     // dynamically loaded cluster
 
-    const pluginRoutesWithParents: any[] = [];
     const orderedPluginRoutes: any[] = [];
 
     // separate plugin routes that have parent and not
     newRoutes.forEach((r: any) => {
       let foundParentRoute;
 
       if (r.parent) {
-        foundParentRoute = allRoutes.find((route) => route.name === r.parent);
+        foundParentRoute = this.findInNestedRoutes(allRoutes, (route: RouteConfig) => route.name === r.parent);
 
         if (foundParentRoute) {
-          pluginRoutesWithParents.push(r);
+          foundParentRoute.children = foundParentRoute?.children || [];
+          foundParentRoute.children.unshift(r.route);
         }
       }
 
       if (!foundParentRoute) {
-        orderedPluginRoutes.push(r.route);
+        orderedPluginRoutes.unshift(r.route);
```

Comment:
So routing order matters in vue-router. We need to push the more specific routes for plugins to be before the default or the default will match first. I'll take a look at this elemental issue.

Reply:
@jordojordo you were right about the cause. We still want to push the new routes to the front but because we were pushing them one at a time we changed the order which caused the issue you illustrated here.

We wanted:
`[...[new routes], ...[old routes]]`

We got:
`[...[setuor wen], ...[old routes]]`